wordpress-mobile/WordPress-Android
# Remove login subtree

Removes login subtree in favor of a binary dependency of https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/.

In https://github.com/wordpress-mobile/WordPress-Android/pull/14707, https://github.com/woocommerce/woocommerce-android/pull/4081 & https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/57, we have prepared all 3 repositories for this removal and in https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/58 we have added publishing to s3 step to the library.

This PR:
1. Removes login library from lint and lint-baseline 8b5be37693869c54d624017124059b38eef34c6d
2. Removes `libs/login` folder and adds the login library as a binary dependency 5a927109ba976beca31778083832d1cfe8698cb0 & 8b1a3133de16a948e286d5225cb8400eae169e84
3. Updates translation handling in https://github.com/wordpress-mobile/WordPress-Android/pull/14778 & https://github.com/wordpress-mobile/WordPress-Android/pull/14791

I've verified that there are no changes to the `libs/login` since https://github.com/wordpress-mobile/WordPress-Android/pull/14707 except for Android Gradle Plugin version update which I'll take care of in the library when I add composite build support. https://github.com/wordpress-mobile/WordPress-Android/commits/develop/libs/login

**To test:**
* Test login/signup flows - I think a smoke test is enough to get this PR merged since we didn't have any issues with switching from subtree to binary dependency in the past. We can make an announcement once the PRs are merged and ask it to be tested and in the unlikely case that there is something broken, it should be easy to address it or revert this PR.

## Regression Notes
1. Potential unintended areas of impact
* Broken login flows

2. What I did to test those areas of impact (or what existing automated tests I relied on)
* Manually tested login/signup flows and existing automated login connected tests

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
